cluster-autoscaler/cloudprovider/azure/azure_manager.go

```diff
@@ -316,6 +316,9 @@ func (m *AzureManager) Refresh() error {
 func (m *AzureManager) forceRefresh() error {
 	if err := m.fetchAutoAsgs(); err != nil {
```

Comment:
we should refactor this logic out. This does a List scale and can possibly be merged with the `Nodes()` call - only affects autodiscovery though.

Reply:
makes sense, could you add a TODO for it? we could do later after this PR.